src/engraving/dom/staff.h

```diff
@@ -310,7 +312,25 @@ class Staff final : public EngravingItem
 
     std::map<int, int> m_channelList[VOICES];
     std::map<int, SwingParameters> m_swingList;
-    std::map<int, CapoParams> m_capoMap;
+
+    class SelectionKeeper
+    {
+    public:
+        SelectionKeeper(Staff* s, ElementType type);
+        ~SelectionKeeper();
+
+    private:
+        Staff* m_staff = nullptr;
+        Score* m_score = nullptr;
+        std::vector<EngravingItem*> m_selectedElements;
+    };
+
+    struct CapoState {
+        CapoParams params;
+        bool needUpdate = false;
+        bool ignorePrevious = false; // If just added update according to previous capo if exists
```

Comment:
Similar comment here. These two booleans look like information that should be retrieved/constructed during the editing operations, not information that should to be stored as a data member